profiles/schema.py

```diff
@@ -172,6 +176,62 @@ def resolve_total_count(self, info, **kwargs):
         return self.iterable.model.objects.count()
 
 
+class PrimaryContactInfoOrderingFilter(OrderingFilter):
+    """Custom ordering filter
+
+    This filter enables ordering profiles by primary contact info fields, for example by city of primary address.
+    """
+
+    # custom field definitions:
+    # 0. custom field name (camel case format)
+    # 1. field display text
+    # 2. model with foreign key profile_id
+    # 3. field name of the related model
+
+    FIELDS = (
+        ("primaryCity", "Primary City", Address, "city"),
+        ("primaryPostalCode", "Primary Postal Code", Address, "postal_code"),
+        ("primaryAddress", "Primary Address", Address, "address"),
+        ("primaryCountryCode", "Primary Country Code", Address, "country_code"),
+        ("primaryEmail", "Primary Email", Email, "email"),
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # add ascending and descending custom orderings
+        self.extra["choices"] += [(item[0], item[1]) for item in self.FIELDS]
+        self.extra["choices"] += [
+            (f"-{item[0]}", f"{item[1]} (Descending)") for item in self.FIELDS
+        ]
+
+    def filter(self, qs, values):
+        def _get_field_data_by_field(fields, field):
+            for f in fields:
+                if f[0] == field:
+                    return f
+            return None
```

Comment:
you could probably avoid some loops and chain if you turn `FIELDS` into a dict:
```python
FIELDS = {
    "primaryCity": {"label": "Primary City", "model": Address, "field_name": "city"},
    "primaryPostalCode": {"label": "Primary Postal Code", "model": Address, "field_name": "postal_code"},
    "primaryAddress": {"label": "Primary Address", "model": Address, "field_name": "address"},
    "primaryCountryCode": {"label": "Primary Country Code", "model": Address, "field_name": "country_code"},
    "primaryEmail": {"label": "Primary Email", "model": Email, "field_name": "email"},
}

def __init__(self, *args, **kwargs):
    super().__init__(*args, **kwargs)
    # add ascending and descending custom orderings
    self.custom_ordering_options = []
    for field, field_data in self.FIELDS.items():
        self.extra["choices"] += [(field, field_data["label"]), (f"-{field}", f"{field_data["label"]} (Descending)")
        self.custom_ordering_options.extend([field, f"-{field}"])
```
and then you could do `if value in self.custom_ordering_options` or somehting 🤔 
not sure if this looks any better or actually adds readability, you are free to keep your implementation or take some parts from this sugegstions